malaria24/ona/tests/test_models.py

```diff
@@ -386,6 +398,43 @@ def test_facility_name(self):
         self.assertEqual(case2.facility_names, 'Unknown')
 
 
+class JembiReportedCaseTest(MalariaTestCase):
+
+    def setUp(self):
+        super(JembiReportedCaseTest, self).setUp()
+        post_save.disconnect(
+            new_case_alert_case_investigators, sender=ReportedCase)
+        post_save.disconnect(
+            new_case_alert_mis, sender=ReportedCase)
+        post_save.disconnect(
+            new_case_alert_ehps, sender=ReportedCase)
+        '''self.complete_url = ("{}?criteria=value:{}"
+                             .format('http://jembi.org/malaria24'))'''
+
+    def tearDown(self):
+        super(JembiReportedCaseTest, self).tearDown()
+        post_save.connect(
+            new_case_alert_case_investigators, sender=ReportedCase)
+        post_save.connect(
+            new_case_alert_mis, sender=ReportedCase)
+        post_save.connect(
+            new_case_alert_ehps, sender=ReportedCase)
+
+    @responses.activate
+    def test_compile_data(self):
+        case = self.mk_case(first_name="John", last_name="Day", gender="male",
+                            msisdn="0711111111", landmark_description="None",
+                            id_type="said", case_number="20171214-123456-42",
+                            abroad="No", locality="None",
+                            reported_by="+27721111111",
+                            sa_id_number="5608071111083",
+                            landmark="School", facility_code="123456")
+        case.save()
+        case.digest = None
+        d = case.get_data()
+        self.assertEqual(case.get_data(), d)
```

Comment:
This looks like it is just comparing `case.get_data()` to `case.get_data()`.

I think it would be better to compare each field:
`self.assertEqual(d['first_name'], 'John')`

Then you know the dictionary is in the format you expect.

Reply:
Absolutely see what you mean. I'll check the fields to ensure it's correct